frontend/cypress/e2e/spec.cy.ts

```diff
@@ -82,3 +82,57 @@ describe("Testing Create Customer and Attach Subscription", () => {
     cy.get(".ant-table-row").first().click();
   });
 });
+
+describe("Testing Event Tracking Details On Metrics Page", () => {
+  it("Test event details on Metrics page", () => {
+    Login();
+    cy.visit("http://localhost:3000/settings/developer-settings");
+    cy.contains("API Keys");
+    cy.get(".ant-btn.ant-btn-primary").first().click();
+    cy.contains("Create API Key");
+    const apiKeyName = getId(16);
+    cy.get(".ant-input").type(apiKeyName);
+    cy.get(".ant-input[type=text]").should("have.value", apiKeyName);
+    cy.get("[placeholder='Select date']").click();
+    cy.get(".ant-picker-cell-today").siblings().last().click();
+    cy.get(".ant-btn.ant-btn-primary.ant-btn-sm").click();
+    cy.get(".ant-modal-footer .ant-btn.ant-btn-primary").click();
+    cy.contains("Your new key is:");
+    const eventName = "api_call";
+    const date = new Date();
+    const idempotencyId = getId(16);
+    const customerId = getId(16);
+    cy.get(".text-lg.font-main .ant-input").then(($input) => {
+      const apiKey = $input.val();
+      cy.get(".ant-modal-footer .ant-btn.ant-btn-primary").last().click();
+      cy.request({
+        url: "http://localhost:8000/api/track/",
+        method: "POST",
+        headers: {
+          "X-API-KEY": apiKey,
+          "Content-Type": "application/json"
+        },
+        body: {
+          "batch": [{
+            "event_name": eventName,
+            "properties": {
+              "region": "US"
+            },
+            "time_created": date,
+            "idempotency_id": idempotencyId,
+            "customer_id": customerId
+          }]
+        }
+      });
+    });
+    cy.visit("http://localhost:3000/metrics");
+    cy.contains("Create Metric");
+    cy.get(".ant-collapse-header").first().click();
+    cy.contains("event_name").siblings().should("include.text", eventName);
+    cy.contains("customer_id").siblings().should("include.text", customerId);
+    cy.contains("ID").siblings().should("include.text", idempotencyId);
+    const dateString = date.toLocaleString("en-ZA").replace(",", "");
```

Comment:
The `en-ZA` locale seemed to match the date format displayed in the UI the closest, but I can change this if there are better options.

Reply:
should be fine here